src/sagemaker/jumpstart/types.py

```diff
@@ -940,10 +1004,340 @@ def to_json(self) -> Dict[str, Any]:
+        return json_obj
+
+    @property
+    def resolved_config(self) -> Dict[str, Any]:
+        """Returns the final config that is resolved from the list of components.
+
+        Construct the final config by applying the list of configs from list index,
+        and apply to the base default fields in the current model specs.
+        """
+        if self.resolved_metadata_config:
+            return self.resolved_metadata_config
+
+        resolved_config = JumpStartMetadataBaseFields(self.base_fields)
+        for component in self.config_components.values():
+            resolved_config = deep_override_dict(
+                deepcopy(resolved_config.to_json()),
+                deepcopy(component.to_json()),
+                component.OVERRIDING_DENY_LIST,
+            )
+        self.resolved_metadata_config = resolved_config
+
+        return resolved_config
+
+
+class JumpStartMetadataConfigs(JumpStartDataHolderType):
+    """Data class to hold the set of JumpStart Metadata configs."""
+
+    __slots__ = ["configs", "config_rankings", "scope"]
+
+    def __init__(
+        self,
+        configs: Optional[Dict[str, JumpStartMetadataConfig]],
+        config_rankings: Optional[Dict[str, JumpStartConfigRanking]],
+        scope: JumpStartScriptScope = JumpStartScriptScope.INFERENCE,
+    ):
+        """Initializes a JumpStartMetadataConfigs object.
+
+        Args:
+            configs (Dict[str, JumpStartMetadataConfig]):
+                List of configs that the current model has.
+            config_rankings (JumpStartConfigRanking):
+                Config ranking class represents the ranking of the configs in the model.
+            scope (JumpStartScriptScope):
+                The scope of the current config (inference or training)
+        """
+        self.configs = configs
+        self.config_rankings = config_rankings
+        self.scope = scope
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartMetadataConfigs object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+    def get_top_config_from_ranking(
+        self,
+        ranking_name: str = JumpStartConfigRankingName.DEFAULT,
+        instance_type: Optional[str] = None,
+    ) -> JumpStartMetadataConfig:
+        """Gets the best the config based on config ranking."""
+        if self.configs and (
+            not self.config_rankings or not self.config_rankings.get(ranking_name)
+        ):
+            raise ValueError("Config exists but missing config ranking.")
+
+        if self.scope == JumpStartScriptScope.INFERENCE:
+            instance_type_attribute = "supported_inference_instance_types"
+        elif self.scope == JumpStartScriptScope.TRAINING:
+            instance_type_attribute = "supported_training_instance_types"
+        else:
+            raise ValueError(f"Unknown script scope {self.scope}")
```

Comment:
Maintain code consistency as a first priority, but here we could consider `NotImplementedError`.